### PR TITLE
auth-4.3.x: Optimize IXFR-to-AXFR fallback path

### PR DESCRIPTION
**Pull request by zeha**
### Short description

Backport of #9163 to 4.3.x.

Avoid making new backends when we are going to either deny the XFR, or
fall back to AXFR anyway.

This cuts down the number of new backends from four (three for IXFR
pre-checks plus one for AXFR) to one (just the AXFR one).
When replying in IXFR mode, we keep making _one_ new backend, which is
also better than before.

While we now hold the s_plock for a while longer, we only take it once
in doIXFR; before we took it twice -- for TSIG retrieval, which now
re-uses the IXFR backend.

### Checklist

I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled this code
- [x] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)
- [ ]  checked that this code was merged to master

### CODE REVIEW

**Comment by zeha** [comment]
Addresses #7795 for 4.3.x. Note: #9163 is not merged yet.

File: pdns/tcpreceiver.cc [hunk 1]
```diff
@@ -1111,48 +1113,42 @@ int TCPNameserver::doIXFR(std::unique_ptr<DNSPacket>& q, int outsock)
...
-  if (!rfc1982LessThan(serial, calculateEditSOA(sd.serial, dk, sd.qname))) {
+  if (serialPermitsIXFR) {
+    DNSName target = q->qdomain;
     TSIGRecordContent trc;
     DNSName tsigkeyname;
     string tsigsecret;
 
+    UeberBackend db;
+    DNSSECKeeper dk(&db);
+    DNSSECKeeper::clearCaches(target);
```
Comment (mind04) [inline pdns/tcpreceiver.cc:1142]: I don't think it is necessary to clean the caches here. There is an other DNSSECKeeper::clearCaches();  a few lines up.

Comment (zeha) [reply pdns/tcpreceiver.cc:1142]: right. fixed this in #9163, will cherry-pick here once 9163 is merged

Comment (Habbie) [reply pdns/tcpreceiver.cc:1142]: 9163 is merged

Comment (zeha) [reply pdns/tcpreceiver.cc]: cherry-picked here

**Pull request merged**